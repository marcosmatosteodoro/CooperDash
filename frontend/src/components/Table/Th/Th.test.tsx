import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { Th } from './'

describe('Th Component in Table Context', () => {
  const testHeader = 'Test Header';
  const testClassName = 'test-class';

  const renderThInTable = () => {
    return render(
      <table>
        <thead>
          <tr>
            <Th header={testHeader} className={testClassName} />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Test cell</td>
          </tr>
        </tbody>
      </table>
    );
  };

  it('renderiza corretamente dentro de uma estrutura de tabela', () => {
    renderThInTable();
    expect(screen.getByRole('columnheader')).toBeInTheDocument();
  });

  it('exibe o texto do cabeÃ§alho correto no contexto da tabela', () => {
    renderThInTable();
    expect(screen.getByText(testHeader)).toBeInTheDocument();
  });

  it('aplica className corretamente dentro da tabela', () => {
    renderThInTable();
    const thElement = screen.getByRole('columnheader');
    expect(thElement).toHaveClass(testClassName);
  });

  it('tem estrutura HTML correta na tabela', () => {
    const { container } = renderThInTable();
    const thElement = container.querySelector('th');
    expect(thElement).toBeInTheDocument();
    expect(thElement?.textContent).toBe(testHeader);
  });
});