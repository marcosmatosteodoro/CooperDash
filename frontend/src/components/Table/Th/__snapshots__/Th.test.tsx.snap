// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Th Component in Table Context corresponde ao instant√¢neo no contexto da tabela 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th
          class="test-class"
        >
          Test Header
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Test cell
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Th Component in Table Context matches snapshot in table context 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th
          class="test-class"
        >
          Test Header
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Test cell
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
